### DISCUSSION TITLE
[d3-selection] v3.0.6 triggers TS-2589 "Type instantiation is excessively deep and possibly infinite"

### DISCUSSION THREAD

**Discussion by david-cho-lerat-sonarsource** [DefinitelyTyped/DefinitelyTyped#66603] [Issues with a @types package]
Hello, we had no issue with @types/d3-selection v3.0.5, but updating to v3.0.6 triggers a TypeScript error on a Selection.call() : TS-2589 "Type instantiation is excessively deep and possibly infinite".

This seems linked to the change introduced by this commit: d172d3e076f25c468c0177896d9369f23c45259c
 "Merge PR #64492 d3-selection: add generic func parameters to selection.call by @ambar-arkin"
 
which introduces this change: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/d172d3e076f25c468c0177896d9369f23c45259c#diff-963d21dc191b8dbf76cd04b065a2a13a485f1e8086b7f31d5d96aa3bb001f3adR882

Any advice on this? Thanks

[a comment by an automated account is not shown]

**Comment by Methuselah96** [comment]
Can you create a TS playground reproducing the issue?
